source/core/Variants.cpp

```diff
@@ -83,65 +106,256 @@ void VariantType::SetVariantToDataTypeError(TypeRef inputType, TypeRef targetTyp
         errPtr->SetErrorAndAppendCallChain(true, kVariantIncompatibleType, "Variant To Data");
     }
     if (outputType && outputData) {
-            outputType->InitData(outputData);
+        outputType->ClearData(outputData);
```

Comment:
The reason InitData and ClearData are separated this way is to save freeing and allocating pointer to VariantData itself for named variables. Otherwise it is unnecessary deallocation for which we are need to allocate again. We can discuss this in person tomorrow in detail if you still have questions.